<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Voxelization</class>
 <widget class="qMRMLWidget" name="Voxelization">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>790</width>
    <height>511</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Inputs</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="labelInput">
        <property name="text">
         <string>Input model:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkSliderWidget" name="pitchWidget">
        <property name="toolTip">
         <string>Set the pitch of the voxels.</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>500.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>imageThreshold</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="inputSelector">
        <property name="toolTip">
         <string>Pick the input to the algorithm.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLModelNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>false</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>inputModel</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelPitch">
        <property name="text">
         <string>Pitch:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="voxelButton">
       <property name="toolTip">
         <string>Run the voxelization.</string>
        </property>
        <property name="text">
         <string>Apply Voxelization</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="outputsCollapsibleButton">
     <property name="text">
      <string>Outputs</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="labelOutputModel">
        <property name="text">
         <string>Output model:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="outputSelectorModel">
        <property name="toolTip">
         <string>Pick the output to the algorithm.</string>
        </property>
        <property name="nodeTypes">
         <stringlist notr="true">
          <string>vtkMRMLModelNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="addEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>true</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>outputModel</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="ctkDirectoryButton" name="DirectoryButton"/>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="exportToFileButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>Export the output model as .vtk .stl .msh</string>
        </property>
        <property name="text">
         <string>Export files to folder</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="advancedCollapsibleButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="text">
      <string>Advanced</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="3">
       <widget class="QPushButton" name="scaleButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>Perform the model scaling.</string>
        </property>
        <property name="text">
         <string>Apply Scaling</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="labelScale">
        <property name="text">
         <string>Scale:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="voxelCountLabel">
        <property name="text">
         <string>Voxel count:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="lineEdit">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="ctkSliderWidget" name="scaleWidget">
        <property name="toolTip">
         <string>Set the value used to scale the input model in size.</string>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>500.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
        <property name="SlicerParameterName" stdset="0">
         <string>imageThreshold</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkDirectoryButton</class>
   <extends>QWidget</extends>
   <header>ctkDirectoryButton.h</header>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>Voxelization</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>inputSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>132</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Voxelization</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>outputSelectorModel</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
